src/coreclr/debug/ee/debugger.cpp

```diff
@@ -5466,6 +5466,12 @@ bool Debugger::FirstChanceNativeException(EXCEPTION_RECORD *exception,
     CONTRACTL_END;
 
 
+    if ((g_fEEShutDown & ShutDown_Finalize2) == ShutDown_Finalize2)
```

Comment:
How does this behave if the case that triggered this is an int3/bp that we set? Will we race to crash with the runtime that's shutting down?

Reply:
> Skip over the breakpoint as normal but potentially forego delivering the cross-process debugger notification.

I think this is the best way to fix the issue.

> Remove all breakpoint patches at an earlier point in shutdown so that no exception is generated in the first place

We would need to ensure that no threads are executing the patches before proceeding. To do that, we would need to suspend the runtime that can come with a new set of issues. In general, we try to minimize amount of code that runs during shutdown. We just try to do the absolute minimum required to allow shutdown to progress. The first option fits the bill better.

Reply:
I'm fine with the first option, having two was mostly to afford some flexibility in case one ran into unforeseen issues. Agreed that the 2nd option requires more deliberate work in the shutdown code path.